lib/ansible/modules/cloud/amazon/ec2_region_facts.py

```diff
@@ -0,0 +1,125 @@
+                    'version': '1.0'}
+
+DOCUMENTATION = '''
+---
+module: ec2_region_facts
+short_description: Gather facts about ec2 regions in AWS.
+description:
+    - Gather facts about ec2 regions in AWS.
+version_added: "2.3"
+author: "Henrique Rodrigues (github.com/Sodki)"
+options:
+  filters:
+    description:
+      - A dict of filters to apply. Each dict item consists of a filter key and a filter value. See \
+      U(https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeRegions.html) for \
+      possible filters. Filter names and values are case sensitive. You can also use underscores (_) \
+      instead of dashes (-) in the filter keys, which will take precedence in case of conflict.
+    required: false
+    default: {}
+
+extends_documentation_fragment:
+    - aws
+    - ec2
+'''
+
+EXAMPLES = '''
+# Note: These examples do not set authentication details, see the AWS Guide for details.
+
+# Gather facts about all regions
+- ec2_region_facts:
+
+# Gather facts about a single region
+- ec2_region_facts:
+    filters:
+      region-name: eu-west-1
+'''
+
+RETURN = '''
+regions:
+    description: Regions that match the provided filters. Each element consists of a dict with all the information related to that region.
+    type: list
+    sample:
+'''
+
+
+try:
+    import boto3
+    from botocore.exceptions import ClientError
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+import traceback
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(
+        dict(
+            filters=dict(default={}, type='dict')
+        )
+    )
+
+    module = AnsibleModule(argument_spec=argument_spec)
+
+    if not HAS_BOTO3:
+        module.fail_json(msg='boto3 required for this module')
+
+    region, ec2_url, aws_connect_params = get_aws_connection_info(module, boto3=True)
+
+    if region:
+        connection = boto3_conn(
+            module,
+            conn_type='client',
+            resource='ec2',
+            region=region,
+            endpoint=ec2_url,
+            **aws_connect_params
+        )
+    else:
+        module.fail_json(msg="region must be specified")
+
+    # Replace filter key underscores with dashes, for compatibility
+    sanitized_filters = module.params.get("filters")
+    for key in sanitized_filters:
+        sanitized_filters[key.replace("_", "-")] = sanitized_filters.pop(key)
+
+    try:
+        regions = connection.describe_regions(
+            Filters=ansible_dict_to_boto3_filter_list(sanitized_filters)
+        )
+    except ClientError as e:
+        module.fail_json(msg=e.message, exception=traceback.format_exc(e))
+
+    # Turn the boto3 result into ansible_friendly_snaked_names
+    snaked_regions = []
+    for region in regions['Regions']:
+        snaked_regions.append(camel_dict_to_snake_dict(region))
+
+    module.exit_json(regions=snaked_regions)
+
+
+from ansible.module_utils.basic import *
+from ansible.module_utils.ec2 import *
```

Comment:
Please change module imports to the new style. Take a look at [vmware_guest.py](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/vmware/vmware_guest.py) for an example.